lib/Basic/FileInfo.cpp

```diff
@@ -14,22 +14,93 @@
 
 #include <cassert>
 #include <cstring>
+
+#if defined(_WIN32)
+#include <windows.h>
+#include <direct.h>
+#else
 #include <sys/stat.h>
+#endif
 
 using namespace llbuild;
 using namespace llbuild::basic;
 
 bool FileInfo::isDirectory() const {
-  return mode & S_IFDIR;
+  return (mode & S_IFDIR) != 0;
 }
 
+#if defined(_WIN32)
+static void FileTimeToTimestamp(FILETIME fileTime, FileTimestamp *stamp) {
+  // Windows counts file times in 100-nanosecond intervals since January 1, 1601
+  // (UTC)
+  // UNIX counts file times in seconds since January 1, 1970 (UTC)
+  const uint64_t HUNDREDS_OF_NANOSECONDS_PER_SECOND = 10000000;
+  const uint64_t EPOCH_DIFFERENCE =
+    11644473600 * HUNDREDS_OF_NANOSECONDS_PER_SECOND;
+
+  uint64_t totalHundredsOfNanoseconds =
+    (((uint64_t)fileTime.dwHighDateTime << 32) |
+    (uint64_t)fileTime.dwLowDateTime);
+  totalHundredsOfNanoseconds -= EPOCH_DIFFERENCE;
+
+  stamp->seconds =
+    totalHundredsOfNanoseconds / HUNDREDS_OF_NANOSECONDS_PER_SECOND;
+  stamp->nanoseconds =
+    totalHundredsOfNanoseconds % HUNDREDS_OF_NANOSECONDS_PER_SECOND;
+}
+#endif
+
 /// Get the information to represent the state of the given node in the file
 /// system.
 ///
 /// \param info_out [out] On success, the important path information.
 /// \returns True if information on the path was found.
 FileInfo FileInfo::getInfoForPath(const std::string& path, bool asLink) {
   FileInfo result;
+
+#if defined(_WIN32)
+  DWORD attributes;
+  if (asLink) {
```

Comment:
I wonder if this is necessary... we use this in cases where the build is attempt to create a symbolic link and needs to examine the attributes of the file representing the link, but IIRC Windows style links are far less useful and I'm not sure if anyone would want to use that in the same way... or if they did, they would probably want it to be done by a Windows specific tool not the general purpose one.

I'm not familiar enough with Windows-style links to know if this will be useful elsewhere, do you think it will?